### PR TITLE
loading boardId from S3 foe each endpoints

### PR DESCRIPTION
**Pull request by luky89**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -14,6 +14,7 @@
   },
   "homepage": "https://github.com/omatija/Matrix_app_Jira#readme",
   "dependencies": {
+    "aws-sdk": "^2.651.0",
```
Comment (omatija) [inline package.json:17]: update existing version or delete this new one if it is not necessary

File: services/utils.service.js [hunk 2]
```diff
@@ -25,16 +25,31 @@ function loadSettings() {
...
 }
 
-function getJiraBoardIdForTeam(team) {
-    let lookup = settings.teams.find(t => t.teamName == team);
-    return lookup != null ? lookup.jiraBoardId : null;
+function getJiraBoardIdForTeam(client, team) {
+    return loadSettings().then(settings3 => {
+        let teams = settings3.clients[client].teams;
+        let boardId
+        teams.forEach(element => {
```
Comment (omatija) [inline services/utils.service.js:32]: let's use old logic for fetching boardId

File: controllers/estimation.controller.js [hunk 3]
```diff
@@ -2,10 +2,15 @@ const estimService = require('../services/estimation.service')
 const timeUtilService = require('../services/timeUtil.service')
 
 function getEstimation(req, res, next) {
+    let client = req.query.client;
```
Comment (omatija) [inline controllers/estimation.controller.js:5]: this may change to header fetching - TBD

Comment (luky89) [reply controllers/estimation.controller.js:5]: i raised question in ms teams we will see.

File: services/utils.service.js (hunk 2 above)
Comment (luky89) [reply services/utils.service.js:32]: updated

File: services/throughput.service.js [hunk 4]
```diff
@@ -4,30 +4,30 @@ const lsq = require('least-squares')
...
     let troughputs = []
-    let projectKey = ""
-    return jira.getProjectForBoard(boardId)
+    let projectKey = boardId.then(boardId => jira.getProjectForBoard(client, boardId))
         .then(project => {
             projectKey = project.data.values[0].key //TODO iterate over array make projectKey "OR project OR project"
         })
-        .then(() => jira.getSprintsForBoard(boardId))
+    return Promise.all([boardId, projectKey])
+        .then(boardId => jira.getSprintsForBoard(client, boardId[0]))
```
Comment (omatija) [inline services/throughput.service.js:23]: name of this input param is not valid anymore - it is array of boardId and projectKey not just boardId. Can you please update it's name

File: services/focus.service.js [hunk 5]
```diff
@@ -1,43 +1,37 @@
...
-        })
-        .then(dataArray => {
-            let memberNameMap = {};
-            teamMembers.forEach(member => memberNameMap[member.jiraAccountId] = member.name);
-            let sprintMap = {}
-            simplifiedSprints.forEach(sprint => sprintMap[sprint.sprintId] = sprint.sprintName);
-            return calculateFocusFactor(dataArray, memberNameMap, simplifiedSprints, sprintMap);
-        });
+
+    return Promise.all([memberIdsArray, teamMembers, simplifiedSprints]).then(function (results) {
```
[the remark this answers is not in the document]
Comment (luky89) [reply services/focus.service.js:18]: done

File: services/focus.service.js [hunk 6]
```diff
@@ -1,43 +1,37 @@
...
+        let sprintIssuesPromises = getIssuesWithTeamMembersWorklogs(client, results[2], results[0], true);
```
[the remark this answers is not in the document]
Comment (luky89) [reply services/focus.service.js:19]: done

**Pull request merged**